playground/backend/internal/preparers/preparers_utils/utils.go

```diff
@@ -0,0 +1,158 @@
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package preparers_utils
+
+import (
+	"beam.apache.org/playground/backend/internal/logger"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	indentationReplacement = "$0"
+	emptyLine              = ""
+	GraphFileName          = "graph.dot"
+	pythonGraphCodePattern = "$0# Write graph to file\n$0from apache_beam.runners.interactive.display import pipeline_graph\n$0dot = pipeline_graph.PipelineGraph(%s).get_dot()\n$0with open('%s', 'w') as file:\n$0  file.write(dot)\n"
+	newLinePattern         = "\n"
+)
+
+type PipelineDefinitionType int
+
+const (
+	RegularDefinition PipelineDefinitionType = 0
+	WithDefinition    PipelineDefinitionType = 1
+)
+
+// DefineVars creates empty variables
+func DefineVars() (string, string, error, bool, PipelineDefinitionType) {
+	return emptyLine, emptyLine, errors.New(emptyLine), false, RegularDefinition
+}
+
+// AddGraphToEndOfFile if no place for graph was found adds graph code to the end of the file
+func AddGraphToEndOfFile(spaces string, err error, tempFile *os.File, pipelineName string) {
+	line := emptyLine
+	regs := []*regexp.Regexp{regexp.MustCompile("^")}
+	_, err = Wrap(addGraphCode)(tempFile, &line, &spaces, &pipelineName, &regs)
+}
+
+// ProcessLine process the current line from the file by either:
+// - trying to find the definition of the beam pipeline and save the pipeline name
+// - or trying to find the place where to add the code for the graph saving and adds it to the line
+// after it save the line to the temp file
+func ProcessLine(curLine string, pipelineName *string, spaces *string, regs *[]*regexp.Regexp, tempFile *os.File, err error) (bool, PipelineDefinitionType, error) {
+	done := false
+	definitionType := RegularDefinition
+	if *pipelineName == "" {
+		// Try tempFile find where the beam pipeline name is defined
+		definitionType, err = Wrap(getPipelineName)(tempFile, &curLine, spaces, pipelineName, regs)
+	} else {
+		// Try tempFile find where beam pipeline definition is finished and add code tempFile store the graph
+		_, err = Wrap(addGraphCode)(tempFile, &curLine, spaces, pipelineName, regs)
+		if *regs == nil {
+			done = true
+		}
+	}
+	return done, PipelineDefinitionType(definitionType), err
+}
+
+//getPipelineName looking for a declaration of a beam pipeline and it's name
+func getPipelineName(line, spaces, pipelineName *string, regs *[]*regexp.Regexp) PipelineDefinitionType {
```

Comment:
I think PipelineObject/VarName is better. Do you search for a variable name, doesn't it?